### ISSUE TITLE
[FEA] Make `gdf_replace_nulls` be out-of-place

### ISSUE THREAD

**Issue by jrhemstad** [rapidsai/cudf#1798]
**Is your feature request related to a problem? Please describe.**
`gdf_replace_nulls` deviates from the vast majority of `libcudf` operations in that it does not treat the input column as immutable and instead performs the null-replacement operation in place.

https://github.com/rapidsai/cudf/blob/branch-0.8/cpp/include/cudf/functions.h#L1054

`col_out` is modified in-place to replace it's null elements with values from `col_in`.

Having in-place operations is inconsistent with the rest of libcudf. Furthermore, it is inefficient as the Python side is forced to deep-copy the input column before calling `gdf_replace_nulls`.

**Describe the solution you'd like**

While transitioning `gdf_replace_nulls` to a [C++ API](https://github.com/rapidsai/cudf/blob/branch-0.8/cpp/docs/TRANSITIONGUIDE.md), it should also be updated to be out of place.

I imagine an API something like this: 

```
namespace cudf{
gdf_column replace_nulls(gdf_column const& input, gdf_column const& replacement_values)
}
```

This will return a copy of `input`, with null values replaced.

**Comment by kkraus14** [comment]
@shwina was this resolved by #1855?

**Comment by shwina** [comment]
Yes
